WordPress/src/main/java/org/wordpress/android/viewmodel/pages/CreatePageListItemLabelsUseCase.kt

```diff
@@ -0,0 +1,143 @@
+                pageConflictResolver.hasUnhandledAutoSave(pagePostModel)
+        )
+    }
+
+    private fun getLabels(
+        pagePostStatus: PostStatus,
+        isLocalDraft: Boolean,
+        isLocallyChanged: Boolean,
+        uploadUiState: PostUploadUiState,
+        hasUnhandledConflicts: Boolean,
+        hasAutoSave: Boolean
+    ): List<UiString> {
+        val labels: MutableList<UiString> = ArrayList()
+        when {
+            uploadUiState is UploadFailed -> {
+                getErrorLabel(uploadUiState, pagePostStatus)?.let { labels.add(it) }
+            }
+            uploadUiState is UploadingPost -> if (uploadUiState.isDraft) {
+                labels.add(UiStringRes(R.string.page_uploading_draft))
+            } else {
+                labels.add(UiStringRes(R.string.page_uploading))
+            }
+            uploadUiState is UploadingMedia -> labels.add(UiStringRes(R.string.uploading_media))
+            uploadUiState is UploadQueued -> labels.add(UiStringRes(R.string.page_queued))
+            uploadUiState is UploadWaitingForConnection -> {
+                when (uploadUiState.postStatus) {
+                    UNKNOWN, PUBLISHED -> labels.add(UiStringRes(R.string.page_waiting_for_connection_publish))
+                    PRIVATE -> labels.add(UiStringRes(R.string.page_waiting_for_connection_private))
+                    PENDING -> labels.add(UiStringRes(R.string.page_waiting_for_connection_pending))
+                    SCHEDULED -> labels.add(UiStringRes(R.string.page_waiting_for_connection_scheduled))
+                    DRAFT -> labels.add(UiStringRes(R.string.page_waiting_for_connection_draft))
+                    TRASHED -> AppLog.e(
+                            PAGES,
+                            "Developer error: This state shouldn't happen. Trashed pages is in " +
+                                    "UploadWaitingForConnection state."
+                    )
+                }
+            }
+            hasUnhandledConflicts -> labels.add(UiStringRes(R.string.local_page_is_conflicted))
+            hasAutoSave -> labels.add(UiStringRes(R.string.local_page_autosave_revision_available))
+        }
+
+        // we want to show either single error/progress label or 0-n info labels.
+        if (labels.isEmpty()) {
+            if (isLocalDraft) {
+                labels.add(UiStringRes(R.string.page_local_draft))
+            } else if (isLocallyChanged) {
+                labels.add(UiStringRes(R.string.page_local_changes))
+            }
+            if (pagePostStatus == PRIVATE) {
+                labels.add(UiStringRes(R.string.page_status_page_private))
+            }
+            if (pagePostStatus == PENDING) {
+                labels.add(UiStringRes(R.string.page_status_pending_review))
+            }
+        }
+        return labels
+    }
+
+    private fun getErrorLabel(uploadUiState: UploadFailed, pagePostStatus: PostStatus): UiString? {
```

Comment:
I have only replaced `R.string.xyz` so they point to `page` strings. I haven't change anything else in this method.